cli/src/pcluster/config/cluster_config.py

```diff
@@ -1886,7 +1886,10 @@ def __init__(self, instances: List[FlexibleInstanceType], **kwargs):
     @property
     def instance_types(self) -> List[str]:
         """Return list of instance type names in this compute resource."""
-        return [flexible_instance_type.instance_type for flexible_instance_type in self.instances]
+        return [
+            flexible_instance_type.instance_type
+            for flexible_instance_type in self.instances  # pylint: disable=not-an-iterable
+        ]
 
```

Comment:
Rather than doing this, we could consider adding a typing hint at the top of the `SlurmFlexibleComputeResource` class:

```python
class SlurmFlexibleComputeResource(_BaseSlurmComputeResource):
    """Represents a Slurm Compute Resource with Multiple Instance Types."""
    instances: List[FlexibleInstanceType]
    ...
```